x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStep.java

```diff
@@ -331,17 +282,7 @@ public boolean equals(Object obj) {
             return false;
         }
         WaitForRolloverReadyStep other = (WaitForRolloverReadyStep) obj;
-        return super.equals(obj)
-            && Objects.equals(maxSize, other.maxSize)
-            && Objects.equals(maxPrimaryShardSize, other.maxPrimaryShardSize)
-            && Objects.equals(maxAge, other.maxAge)
-            && Objects.equals(maxDocs, other.maxDocs)
-            && Objects.equals(maxPrimaryShardDocs, other.maxPrimaryShardDocs)
-            && Objects.equals(minSize, other.minSize)
-            && Objects.equals(minPrimaryShardSize, other.minPrimaryShardSize)
-            && Objects.equals(minAge, other.minAge)
-            && Objects.equals(minDocs, other.minDocs)
-            && Objects.equals(minPrimaryShardDocs, other.minPrimaryShardDocs);
+        return super.equals(obj) && Objects.equals(conditions, other.conditions);
```

Comment:
Do we still need this `super.equals(obj)`? I don't think we do, right?

Reply:
I believe so too, I am going to double check it though.

Reply:
Hm, it appears we need it, the `WaitForRolloverReadyStep` extends `Step`, the `equals` in `Step` is also checking the `key` and `nextStepKey` of the step, so we should keep it.